libs/MeshKernel/include/MeshKernel/Mesh.hpp

```diff
@@ -319,44 +319,13 @@ namespace meshkernel
         double ComputeMaxLengthSurroundingEdges(UInt node);
 
         /// @brief Build the rtree for the corresponding location, using all locations
-        /// @param[in] meshLocation The mesh location for which the RTree is build
-        void BuildTree(Location meshLocation);
+        /// @param[in] location The mesh location for which the RTree is build
+        void BuildTree(Location location);
 
         /// @brief Build the rtree for the corresponding location, using only the locations inside the bounding box
         /// @param[in] meshLocation The mesh location for which the RTree is build
         /// @param[in] boundingBox The bounding box
         void BuildTree(Location meshLocation, const BoundingBox& boundingBox);
```

Comment:
Isn't this function a little dangerous? We could pass a bounding box that is for a small part of the domain or even outside of the mesh domain, the tree would be empty, and many subsequent operations that used the tree would be invalid